### ISSUE TITLE
Issues that need to be resolved before preview release this week

### ISSUE THREAD

**Issue by jchodera** [openpathsampling/openpathsampling#526]
The issue tracker has a list of 13 issues that are tagged with the `1.0` label:
https://github.com/openpathsampling/openpathsampling/milestone/1

There are also two issues tagged with the `ASAP` label:
https://github.com/openpathsampling/openpathsampling/milestone/3

Do all of these need to be addressed before we get out our preview release this week?

**Comment by dwhswenson** [comment]
> Do all of these need to be addressed before we get out our preview release this week?

`ASAP` issues: Yes. Note that one is under review (I gave it the first round, and it needs some responses from @jhprinz) and the other is probably fixed, but we need to confirm that the bug is actually gone.

`1.0` issues: Not all of them. For example, the reorganization of various directories and miscellaneous renamings of variables should be done before the official release, but we can release a beta without that. Also, 3 issues from that will be closed by #517 (under review, `ASAP` milestone), and (at least) another 2 will be closed by #500 (under review, no milestone). A couple others are related to SRTIS, which we've said doesn't need to be in the beta. Others are things like "cool logo" and "docstring cleanup": good things to put on the final polish, but not required for a working beta.

Note that using the beta could be frustrating since the API might have some name changes. I'll open an issue today with some of the renamings that I think we should consider.

**Comment by jchodera** [comment]
Can we call it an `alpha` and create a `1.0alpha` milestone to flag the issues that we need to address this week?

**Comment by dwhswenson** [comment]
Sure, although it might be easier to just list the main PRs here. I think that we can call it alpha after:

* #517 
* #500 
* #386 

Several of these resolve multiple issues; adding all the issues that they resolve to a `1.0alpha` milestone would probably be *more* confusing, if you're looking at it to see how much work is left.

I gave the first two an initial round of review yesterday. Next step will get done when @jhprinz has time outside his teaching responsibilities this week and responds to the review. Once #517 is merged, I'll fix any incompatibilities in #386, and JHP will review that (as a partially complete PR, just to get the changes to file format included).

Does anyone else see other things as required for an alpha?

**Comment by jchodera** [comment]
@jhprinz is back at 3.00P today, I think, so I'll work with him to see what needs to be done.

**Comment by jchodera** [comment]
@dwhswenson and @jhprinz: Can you give us an update on how far along you are with closing up the three open PRs above so we can cut an alpha?

**Comment by jhprinz** [comment]
From my side:
- #517, this is done and I made changes according to Davids questions. This is a VERY big PR so it might need some more time to check that it works as expected, but otherwise it passes tests and is feature complete
- #500, this one is done and checked by David. There are more associated things we should fix but only for 2.0. This is ready to be merged and I think David is practically done with it.
- #386, needs the other two to be merge first and then we need to apply fixed to merge. I am currently looking at it and I think this is also ready to be merged.

Summary: The PRs are done. Still finishing review and one needs to be fixed when the others are merged.

**Comment by dwhswenson** [comment]
Yep. This morning, all three of these were on JHP's side of the court. Now all three are on mine. I'm about to look over #500; unless there's a surprise, it will be merged tonight. As I mentioned in a comment to #517, because of its size, I'm not going to review the whole thing in detail, but instead trust that it does what JHP says. I'll give it a brief once-over, but will probably merge it either tonight or tomorrow (I think JHP addressed everything from my first round of review). That leaves #386, which I'll need to update to match the recent changes. I probably won't finish that until tomorrow.

I'm also preparing a PR that will have significant effects on some user-facing stuff (this is the treatment of the MS-outer interface that JHP and I discussed in Berlin). I'm hoping that will be ready for review tomorrow night, Wednesday at the latest. This wasn't part of the original 1.0-alpha, but if we're still making some fixes to the paper (I'd like to take a run through before we put a first preprint out), then it might be worth getting that change in and updating the paper to match it.

(Early version of the main file in that PR: [ms_outer_interface.py](https://github.com/dwhswenson/openpathsampling/blob/8af980af6f0a28009d49287bef025370f9986e1d/openpathsampling/high_level/ms_outer_interface.py))

**Comment by jhprinz** [comment]
#500 is done.

**Comment by jchodera** [comment]
> I'm also preparing a PR that will have significant effects on some user-facing stuff (this is the treatment of the MS-outer interface that JHP and I discussed in Berlin). I'm hoping that will be ready for review tomorrow night, Wednesday at the latest. This wasn't part of the original 1.0-alpha, but if we're still making some fixes to the paper (I'd like to take a run through before we put a first preprint out), then it might be worth getting that change in and updating the paper to match it.

Sounds good. Go ahead with this.

I'm still working on my paper PR that makes changes to Abstract and Introduction, along with some structural changes. Hope to have this finished up tonight, and then put the rest of my edits in Wed: https://github.com/choderalab/pathsampling-manuscripts/pull/28

**Comment by dwhswenson** [comment]
Updates on this:

* #500 and #517 have been merged
* #386 is under review by @jhprinz 
* #530 is the PR-in-preparation I referenced above. Core code written, with tests. I'm hoping I'll be able to integrate that into the main code tomorrow, and have it ready to review very soon.

The only other thing I'm aware of that will make stored files incompatible or require changes to the (existing) text of the paper will be the renamings in #527. @jhprinz, is there anything else you have on the way for 1.0 that we need to keep in mind? Otherwise, I think everything is clean-up and polish.

**Comment by jhprinz** [comment]
#386 is mergeable from my point of view. So go ahead and merge or I do it tomorrow. It is still marked as WIP, so I am not sure.

I will try to fix the alanine example stuff tomorrow. Means we can do the renaming session and then release the alpha

**Comment by jchodera** [comment]
Do you need to do a docs cleanup pass too at some point?

**Comment by jhprinz** [comment]
> Do you need to do a docs cleanup pass too at some point?

Yes, definitely. We should do a last pass after finish renaming things. Of course, we can start earlier. We do ironing the docs once in a while but have postponed most of it until the API is done.

**Comment by jhprinz** [comment]
# 386 is merged. So we are basically done with the hard part.

**Comment by jchodera** [comment]
Great!

Let me know when I should put together a conda recipe. You probably want to cut a release at some point. It's totally OK if this is 0.9.1 or something, since we want 1.0 to be the version that goes with the actual paper release.

**Comment by jchodera** [comment]
I would suggest we get into the process of cutting releases more rapidly at this point. Once those PRs are merged, we can cut incremental releases (0.9.x) as different rounds of bugfixes or cleanups are performed.

**Comment by dwhswenson** [comment]
With the merge of #530 (following merges of the PRs I listed originally), the only thing that I think will have API-breaking effects will be renamings in #527. (@jhprinz: you agree?)

This means that, if you don't mind the renamings causing trouble, I think we're ready to cut a 0.9.1 (or so) release, assuming @jhprinz is in agreement. Once we have the renamings set, we could even call it 1.0, although there might be some quick 1.0.1 or 1.0.2 follow-up.

**Comment by jchodera** [comment]
I think we should aim to have a series of 0.9.x releases (marked as "preview" when cutting the github release) until the 1.0.0 release corresponding to paper submission.

**Comment by dwhswenson** [comment]
Sure: so is it `1.0beta` (or `1.0release_candidate`) == submission, with `1.0` (proper) == proofs? That's my gut on how to do this.

In any case, I think that the API changes will stop within the next week. There's plenty of cleanup to do (unused fucntions and better docs, in particular), but we should be reasonably stable.

**Comment by jhprinz** [comment]
Yes, I agree. We can start with an `0.9.1-alpha`. While making renamings and little updates as we finish the example for the paper.

**Comment by jchodera** [comment]
Let's not call them `alpha` or `beta`, since this makes things more complicated and there's no need for a lengthy release candidate testing period. Let's just run up the `0.9.z` counter until we actually submit the paper, and have `1.0.0` be the submitted version (with `1.0.z` being post-submission bugfixes). 

Major changes to the API that break backwards-compatibility should cause us to go from `1.y.z` to `2.y.z`, while major feature updates should increment the `y` from `1.0.z` to `1.1.z`.

**Comment by jhprinz** [comment]
The two PRs are ready. Removal of two files and one renaming. Should not interfere with anything

**Comment by jhprinz** [comment]
Can this be closed? I think all the PR we wanted are merged.

**Comment by dwhswenson** [comment]
Seems to me that releasing 0.9.0 implied this was solved, but what do you think, @jchodera?

**Comment by jchodera** [comment]
Yes!
